core/domain/user_services.py

```diff
@@ -551,3 +556,16 @@ def get_user_impact_score(user_id):
         return model.impact_score
     else:
         return 0
+
+def fetch_gravatar(email):
+    """Returns the gravatar corresponding to given email"""
```

Comment:
Returns the gravatar corresponding to the user's email, or an identicon generated from the email if the gravatar doesn't exist.

Note: if the call to the gravatar service fails, this returns DEFAULT_IDENTICON_DATA_URL and logs an error.